devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy

```diff
@@ -0,0 +1,312 @@
+            state "battery", label: '${currentValue}% battery', unit: ""
+        }
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+        }
+    
+        main(["motion", "temperature"])
+        details(["motion", "temperature", "humidity", "illuminance", "battery", "refresh"])
+    }
+}
+
+private List<Map> collectAttributes(Map descMap) {
+    List<Map> descMaps = new ArrayList<Map>()
+
+    descMaps.add(descMap)
+
+    if (descMap.additionalAttrs) {
+        descMaps.addAll(descMap.additionalAttrs)
+    }
+
+    return  descMaps
+}
+
+def parse(String description) {
+    log.debug "description: $description"
+    Map map = zigbee.getEvent(description)
+    //log.debug "map: $map"
+    if (!map) {
+        if (description?.startsWith('zone status')) {
+            map = parseIasMessage(description)
+        } else if (description?.startsWith('read attr')) {
+            Map descMap = zigbee.parseDescriptionAsMap(description)
+
+            if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
+                //log.info "BATT METRICS - attr: ${descMap?.attrInt}, value: ${descMap?.value}, decValue: ${Integer.parseInt(descMap.value, 16)}, currPercent: ${device.currentState("battery")?.value}, device: ${device.getDataValue("manufacturer")} ${device.getDataValue("model")}"
+                List<Map> descMaps = collectAttributes(descMap)
+                def battMap = descMaps.find { it.attrInt == 0x0020 }
+
+                if (battMap) {
+                    map = getBatteryResult(Integer.parseInt(battMap.value, 16))
+                }
+            } else if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002 && descMap.commandInt != 0x07) {
+                def zs = new ZoneStatus(zigbee.convertToInt(descMap.value, 16))
+                map = translateZoneStatus(zs)
+            } else if (descMap?.clusterInt == zigbee.TEMPERATURE_MEASUREMENT_CLUSTER && descMap.commandInt == 0x07) {
+                if (descMap.data[0] == "00") {
+                    //log.debug "TEMP REPORTING CONFIG RESPONSE: $descMap"
+                    sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+                } else {
+                    log.warn "TEMP REPORTING CONFIG FAILED- error code: ${descMap.data[0]}"
+                }
+            } else if (descMap?.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == zigbee.ATTRIBUTE_IAS_ZONE_STATUS && descMap?.value) {
+                map = translateZoneStatus(new ZoneStatus(zigbee.convertToInt(descMap?.value)))
+            } else if (descMap?.clusterInt == 0x0406 && descMap.attrInt == 0 /*occupancy*/ && descMap?.value) {
```

Comment:
Provide a private getter method to use instead of those hardcoded numbers. Something like here https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/a650b9d970b52ae8cb817a1747f113b7070a1d26/devicetypes/smartthings/leaksmart-water-sensor.src/leaksmart-water-sensor.groovy#L64 
And it'd be preferred to use `0x0000` instead of `0` when comparing to `descMap.attrInt`.